goalc/main.cpp

```diff
@@ -201,6 +176,8 @@ int main(int argc, char** argv) {
     status = ReplStatus::WANT_EXIT;
   }
 
+  // TODO - investigate why there is such a delay when exitting
```

Comment:
If the delay happens only after doing an `(mi)`, it's likely that the destructor for `Compiler` is slow - it has to basically delete/free everything. At least on linux it's only like a second or so, and I've never looked into it.  But if you wanted, you could just `std::abort` to exit really quickly.